Notifications/Notifications/Presentation/Settings/NotificationsSettingsView.swift

```diff
@@ -7,19 +7,68 @@
 
 import SwiftUI
 import Theme
+import Core
 
 public struct NotificationsSettingsView: View {
+    @ObservedObject
+    private var viewModel: NotificationsSettingsViewModel
+    
+    public init(viewModel: NotificationsSettingsViewModel) {
+        self.viewModel = viewModel
+    }
+    
     public var body: some View {
         GeometryReader { proxy in
             ZStack(alignment: .top) {
-                VStack(alignment: .center) {
-                    Text("Notificaion settings view")
+                VStack(alignment: .leading) {
+                    HStack {
+                        Text(NotificationsLocalization.Settings.preferenceTitle)
+                            .font(Theme.Fonts.titleMedium)
+                            .foregroundStyle(Theme.Colors.textPrimary)
+                        
+                        Spacer()
+                        Toggle(isOn: $viewModel.hasPermission, label: {})
+                            .toggleStyle(SwitchToggleStyle(tint: Theme.Colors.toggleSwitchColor))
+                            .frame(width: 50)
+                            .accessibilityIdentifier("discussion_switch")
+                            .highPriorityGesture(
+                                DragGesture(minimumDistance: 20, coordinateSpace: .local).onEnded { _ in
+                                    viewModel.toggleNotificationsPermissionAction()
+                                }
+                            )
+                            .highPriorityGesture(
+                                TapGesture().onEnded {
+                                    viewModel.toggleNotificationsPermissionAction()
+                                }
+                            )
```

Comment:
We added PriorityGesture [here](https://github.com/edx/edx-mobile-marketplace-ios/pull/98) to fix iOS18 bug when parental View intercepts gestures and when we use Toggle not just like 'toggle'. Do you see the same here? Otherwise we don't need `. highPriorityGesture` here. Please look. Thank you